Samples/Widgets/MultipleWidgets/cpp-win32-packaged/SampleWidgetProviderApp/WidgetProvider.cpp

```diff
@@ -0,0 +1,175 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "pch.h"
+#include "WidgetProvider.h"
+#include "WidgetImpl.h"
+
+typedef std::function<std::shared_ptr<WidgetImpl>(winrt::WidgetContext, winrt::hstring)> CreateWidgetImplFn;
+struct WidgetImplCreationInfo
+{
+	CreateWidgetImplFn factoryFn;
+};
+
+template<typename T>
+std::shared_ptr<WidgetImpl> CreateWidgetFn(winrt::WidgetContext widgetContext, winrt::hstring state)
+{
+	auto widgetId = widgetContext.Id();
+	auto newWidget = std::make_shared<T>(widgetId, state);
+
+	// Call UpdateWidget() to send data/template to the newly created widget.
+	winrt::WidgetUpdateRequestOptions updateOptions{ widgetId };
+	updateOptions.Template(newWidget->GetTemplateForWidget());
+	updateOptions.Data(newWidget->GetDataForWidget());
+	// You can store some custom state in the widget service that you will be able to query at any time.
+	updateOptions.CustomState(newWidget->State());
+	// Update the widget
+	winrt::WidgetManager::GetDefault().UpdateWidget(updateOptions);
+	return newWidget;
+}
+
+
+#include "CountingWidgetImpl.h"
+#include "WeatherWidgetImpl.h"
+// Register all widget types here, optionally specify an is enabled function
+static auto s_widgetImplRegistry = std::unordered_map<winrt::hstring, WidgetImplCreationInfo>{
+	// { WIDGET_NAME, { &WIDGET_FACTORYFN }}
+	{ L"Counting_Widget",			{ &CreateWidgetFn<CountingWidget>}},
+	{ L"Weather_Widget",			{ &CreateWidgetFn<WeatherWidget>}},
+};
+
+std::unordered_map<winrt::hstring, std::shared_ptr<WidgetImpl>> WidgetProvider::m_runningWidgetImpl{};
+
+WidgetProvider::WidgetProvider()
+{
+	RecoverRunningWidgets();
+}
+
+std::shared_ptr<WidgetImpl> WidgetProvider::InitializeWidgetInternal(winrt::Microsoft::Windows::Widgets::Providers::WidgetContext widgetContext, winrt::hstring state)
+{
+	auto widgetName = widgetContext.DefinitionId();
+	auto widgetId = widgetContext.Id();
+	if (s_widgetImplRegistry.find(widgetName) != s_widgetImplRegistry.end())
+	{
+		auto widgetImpl = s_widgetImplRegistry[widgetName].factoryFn(widgetContext, state);
+		m_runningWidgetImpl[widgetId] = widgetImpl;
+		return widgetImpl;
```

Comment:
indentation here and in other places seems odd; run clang tidy or similar (VS format document) for consistency

Reply:
identation fixed in code files